### PR TITLE
✨ Feat: #381 Add generic video primitives layer to scene-studio

### LINKED ISSUE #381: [Feature]: Add generic video primitives layer to scene-studio
## Preliminary Checks
- [x] I have searched existing issues and discussions
- [x] This feature doesn't already exist
- [x] This aligns with the project's goals

## Target Application
Other (specify in description) — `apps/scene-studio` (`src/primitives/` layer)

## Feature Type
New component/UI element

## Priority
Medium (would significantly improve the project)

## Feature Summary
Add the layer-1 generic video primitives (VideoTitle, Caption, SafeArea, GradientOverlay, Logo, MediaFrame, BrandOutro) with video typography and motion tokens inside `apps/scene-studio`, so compositions share brand-consistent building blocks.

## Problem Statement
Video compositions need reusable, brand-consistent primitives. Per `.claude/rules/remotion-template-guidelines.md` (added in #380), small parts must be generic — no use-case-, material-, or platform-specific names. Phase 2 of the video platform plan, follow-up to #380.

Design note: the primitives live **inside the app** (`src/primitives/`), not in a `packages/` workspace — `packages/` is reserved for code with multiple consumers, and scene-studio is currently the only one. Extraction to `packages/` happens only when a second consumer (e.g. a Remotion Player web preview) actually exists, mirroring the "commonize after real duplication" template rule.

## Proposed Solution
- `apps/scene-studio/src/primitives/`: `VideoTitle`, `Caption`, `SafeArea`, `GradientOverlay`, `Logo`, `MediaFrame`, `BrandOutro` (named exports, `Props` interface, `className` merged via `cn`), plus `video-theme.css` (`--text-scene-*` typography extension applying size/line-height/weight per class).
- `apps/scene-studio/src/tokens/`: motion tokens (`SCENE_FPS`, frame-based durations, easings, springs) and `SAFE_AREA_INSETS` for 1080×1920 vertical video.
- `apps/scene-studio/src/utils/`: local `cn` with a scene-specific `extendTailwindMerge` (registers `text-scene-*` in the font-size group; importing `ui`'s cn would drag the web module graph into the Remotion bundle and deep `ui/src` imports are being eliminated) + merge regression tests.
- One demo composition per primitive (`<Folder name="primitives">`, IDs `primitive-*`), asset-free.

Out of scope: Storybook / Chromatic (Studio demo compositions are the verification surface), video patterns / schemas (#382).

## Acceptance Criteria
- [ ] All seven primitive demos render and scrub in Remotion Studio
- [ ] `text-scene-*` classes visibly apply inside compositions
- [ ] No Remotion dependency leaks into web packages (`ui`, apps)
- [ ] `pnpm lint && pnpm typecheck && pnpm test && pnpm docs:check` pass

## Technical Requirements
- [x] Requires new dependencies

## Breaking Changes
No

## Additional Context
Depends on #380.

## Contribution
- [x] I'm willing to submit a PR for this feature

### PR DESCRIPTION
**Pull request by krd-knt**
## Summary

Adds the layer-1 **generic video primitives** to `apps/scene-studio` (Phase 2 of the video platform plan): seven use-case-agnostic components, video-scale typography tokens, motion/safe-area tokens, and one demo composition per primitive for verification in Remotion Studio.

### What changed?

- `src/primitives/` — `VideoTitle`, `Caption`, `SafeArea`, `GradientOverlay`, `Logo`, `MediaFrame`, `BrandOutro` (named exports, `Props` interface, `className` merged via `cn`) + `video-theme.css` with `--text-scene-*` typography (each class applies size / line-height / weight via Tailwind v4 sub-variables; sized for 1080×1920 phone-viewed frames)
- `src/tokens/` — `SCENE_FPS`, frame-based `durationsInFrames`, `easings`, `springs`, and `SAFE_AREA_INSETS` (IG Reels / TikTok overlay union: top 220 / right 160 / bottom 440 / left 60)
- `src/utils/` — local `cn` with a scene-specific `extendTailwindMerge` (registers `text-scene-*` in the font-size class group) + merge regression tests
- `src/compositions/primitives/` — seven asset-free demo compositions (`primitive-video-title` … `primitive-brand-outro`) registered under a `primitives` Studio folder
- Docs: AGENTS.md, template guidelines, and app README updated with the layer structure and the extraction criterion

### Why was this changed?

- Compositions need brand-consistent reusable building blocks before the first data-driven pattern lands (#382)
- **Design decision:** the primitives live inside the app (`src/primitives/`), not in a `packages/` workspace — `packages/` is reserved for multi-consumer code and scene-studio is currently the only consumer. Extraction to `packages/` happens when a second consumer (e.g. a Remotion Player web preview) actually exists, mirroring the template guidelines' "commonize after real duplication" rule
- `cn` is intentionally app-local: importing it from `ui` would pull the web component module graph (Base UI / i18next / sonner) into the Remotion bundle via the barrel, and deep `ui/src` imports are being eliminated (#349). Each token set is registered once, next to its owner

## Type of Change

- [x] ✨ New feature (non-breaking change that adds functionality)
- [x] 🧪 Test additions or updates
- [x] 📝 Documentation update
- [x] 📦 Dependency updates

## Affected Applications

Only the new app `apps/scene-studio/`; blog / portfolio / ui and shared packages are untouched.

### Test Coverage

- [x] Unit tests added/updated
- [x] Manual testing completed

`cn.test.ts` covers the `text-scene-*` merge regression risk (custom classes misclassified as colors by default tailwind-merge). Visual components are verified via the Studio demo compositions per the agreed policy (no jsdom render tests, no Storybook).

### How to Test

1. `pnpm install`
2. `pnpm -F scene-studio dev` → open the `primitives` folder in the Studio sidebar and scrub all seven demos
3. `pnpm -F scene-studio render primitive-video-title` → MP4 renders with `text-scene-title` (112px/bold) applied
4. `pnpm lint && pnpm typecheck && pnpm test && pnpm docs:check`

### Test Results

- [x] All existing tests pass (`pnpm test`)
- [x] New tests pass
- [x] Build succeeds (`pnpm build`)
- [x] Linting passes (`pnpm lint`)
- [x] Type checking passes

Verified locally: still-frame renders confirm the video typography theme, SafeArea guides/insets, tone variants, and BrandOutro layout.

## Database Changes

- [x] No database changes

## Environment Variables

- [x] No environment variable changes

### Before

n/a (new components)

### After

`pnpm -F scene-studio render primitive-brand-outro` etc. — see the seven `primitive-*` compositions in Studio.

## Breaking Changes

- [x] No breaking changes

## Deployment Notes

- [x] No special deployment requirements

## Documentation

- [x] Documentation updated (README, CLAUDE.md, etc.)

## Checklist

- [x] Code follows the project's coding standards
- [x] Self-review completed
- [x] Changes are minimal and focused
- [x] Commit messages are clear and descriptive
- [x] Branch is up to date with main
- [x] No console errors or warnings
- [x] Performance impact considered

## Related Issues

Closes #381
Relates to #380, #382

## Additional Context

Issue #381 was retitled/rescoped from "packages/scene-ui" to the app-internal layer after a design review (packages/ = multi-consumer only). The `docs-check-ignore` escape added in #383 for the forward reference was removed here.

## Reviewer Notes

- `src/utils/extendTailwindMerge.ts` mirrors `packages/ui/src/utils/extendTailwindMerge.ts` on purpose (different token sets, same pattern); unifying both into `tailwind-config` is a possible future refactor, out of scope here
- SafeArea insets are a pragmatic union of platform overlay zones — expect tuning after real posts

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: apps/scene-studio/src/primitives/VideoTitle/VideoTitle.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  className,
+}: Props) {
+  const frame = useCurrentFrame();
+  const { fps } = useVideoConfig();
+
+  const enter = spring({
+    frame: frame - enterDelayInFrames,
+    fps,
+    config: springs.gentle,
+  });
```
[the remark this answers is not in the document]
Comment (krd-knt) [reply apps/scene-studio/src/primitives/VideoTitle/VideoTitle.tsx:32]: Addressed in 8d3ded1: extracted the pure logic into co-located, tested modules — `Caption/captionMotion.ts` (enter-delay/exit interpolation, 7 cases), `GradientOverlay/overlayBackground.ts` (position/tone/opacity computation), `MediaFrame/isSelfContainedSource.ts` (source branching incl. edge cases), and `VideoTitle` tone→token mapping. 26 tests now run under `pnpm -F scene-studio test`. Remaining JSX (SafeArea inset pass-through, spring wiring) is declarative Remotion rendering, verified via the `primitive-*` demo compositions per the agreed no-jsdom-render-test policy.

**Comment by krd-knt** [comment]
@codex review

File: apps/scene-studio/src/primitives/MediaFrame/MediaFrame.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+export function MediaFrame({
+  src,
+  mediaType,
+  fit = 'cover',
+  transform,
+  startFromInFrames,
+  muted = true,
+  className,
+}: Props) {
+  const resolvedSource = isSelfContainedSource(src) ? src : staticFile(src);
```
[the remark this answers is not in the document]
Comment (krd-knt) [reply apps/scene-studio/src/primitives/MediaFrame/MediaFrame.tsx:25]: Addressed in e04cc1c: added `MediaFrame.test.tsx` — a jsdom component test that mocks `remotion` (`Img`/`OffthreadVideo`/`staticFile` stubs) and verifies the component's contract: relative sources are resolved through `staticFile()` for both media types, remote URLs pass through untouched, `mediaType` switches between the image and video elements, and trim (`startFrom`) / default `muted` settings are forwarded to `OffthreadVideo`. 30 tests now pass under `pnpm -F scene-studio test`.

**Comment by krd-knt** [comment]
@codex review
